rskj-core/src/main/java/co/rsk/peg/union/UnionBridgeStorageProviderImpl.java

```diff
@@ -107,4 +107,18 @@ public void increaseWeisTransferredToUnionBridge(Coin amountRequested) {
         Coin currentWeisTransferredToUnionBridge = getWeisTransferredToUnionBridge().orElse(Coin.ZERO);
         this.weisTransferredToUnionBridge = currentWeisTransferredToUnionBridge.add(amountRequested);
     }
+
+    @Override
+    public void decreaseWeisTransferredToUnionBridge(Coin amountToRelease) {
+        if (isNull(amountToRelease) || amountToRelease.compareTo(Coin.ZERO) < 0) {
+            throw new IllegalArgumentException("Amount released cannot be null or negative");
+        }
+        Coin currentWeisTransferredToUnionBridge = getWeisTransferredToUnionBridge().orElse(Coin.ZERO);
```

Comment:
```suggestion
        Coin zeroCoins = Coin.ZERO;
        Coin currentWeisTransferredToUnionBridge = getWeisTransferredToUnionBridge().orElse(zeroCoins);
```

wdyt? to avoid duplicated code below

Reply:
I don't consider using `Coin.ZERO` twice as duplicated code, or not a big deal in those terms. I prefer to keep the current implementation.